yt/frontends/ramses/io.py

```diff
@@ -91,28 +91,68 @@ def _read_particle_fields(self, chunks, ptf, selector):
                         data = np.asarray(rv.pop((ptype, field))[mask], "=f8")
                         yield (ptype, field), data
 
+    def _generic_handler(self, fname, foffsets, data_types,
```

Comment:
It looks like this function doesn't actually use `self` and thus doesn't need to be a member function on this class. Can you rearrange this so that `_generic_handler` is a plain function defined above this class?